src/main/java/com/nha/abdm/wrapper/hip/hrp/database/mongo/services/PatientService.java

```diff
@@ -171,6 +172,12 @@ public void addConsent(String abhaAddress, Consent consent) throws IllegalDataSt
       throw new IllegalDataStateException("Patient not found in database: " + abhaAddress);
     }
     List<Consent> consents = patient.getConsents();
+    if (consents == null) {
```

Comment:
this is duplicate code, we can place the for loop inside if check for `consents` not being empty.. 
if (!CollectionUtils.isEmpty(consents)) { for(...) {..} }